code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForApiAuthorizationServlet.java

```diff
@@ -35,17 +35,17 @@
 @WebServlet("/check-for-api-authorization")
 public abstract class CheckForApiAuthorizationServlet extends HttpServlet {
 
-  //access API with the created credential
-  abstract void useCredential(Credential credential);
-  //URI pointing to redirect to the servlet that implements this class
+  // access API with the created credential
+  abstract void useCredential(String userId, Credential credential, HttpServletResponse response) throws IOException, InterruptedException;
+  // URI pointing to redirect to the servlet that implements this class
   abstract String getServletURIName();
-  //Update the userId with the newly created credential
-  abstract void updateUser(String userId);
+  // Update the userId with the newly created credential
+  abstract void updateUser(String userId, String email);
 
   private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
   private static final List<String> SCOPES = Collections.singletonList("https://www.googleapis.com/auth/contacts.readonly");
   private static final String CREDENTIALS_FILE_PATH = "WEB-INF/credentials.json";
-  private static final String url = "https://covid-catchers-fixed-gcp.ue.r.appspot.com";
+  private static final String url = "https://8080-49ecfd50-1d05-462f-af38-ebb02e752a59.us-central1.cloudshell.dev";
```

Comment:
So, I was told this is a debugging one and that the covid-catchers-fixed-gcp.ue.r.appspot.com one is the one you actually want, right?

Reply:
yep, I forgot to put it back